zephyr/esp32c3/src/bt/esp_bt_adapter.c

```diff
@@ -794,18 +861,22 @@ static bool async_wakeup_request(int event)
 
 static void async_wakeup_request_end(int event)
 {
-	bool request_lock = false;
+	if (s_lp_cntl.enable == 0) {
+		return;
+	}
 
+	bool allow_to_sleep;
 	switch (event) {
-	case BTDM_ASYNC_WAKEUP_REQ_HCI:
-	case BTDM_ASYNC_WAKEUP_REQ_CTRL_DISA:
-		request_lock = true;
-		break;
-	default:
-		return;
+		case BTDM_ASYNC_WAKEUP_SRC_VHCI:
+		case BTDM_ASYNC_WAKEUP_SRC_DISA:
+			allow_to_sleep = true;
+			break;
+		default:
+			allow_to_sleep = true;
```

Comment:
Can't see the need for the switch statement here, should be allow_to_sleep = false; on default?

Reply:
Same reply as in S3 BT support. I kept original file functions as is. Suggest not to change that for now.